### ISSUE TITLE
Cannot connect to exchange using v1.0.6

### ISSUE THREAD

**Issue by delfuego** [brandonhamilton/bunyan-logstash-amqp#3]
When using version 1.0.5, the following works perfectly:

``` javascript
var amqpStream = require('bunyan-logstash-amqp').createStream({
  host: 'mq1.example.com',
  login: 'oit-scribe',
  password: 'redacted',
  vhost: '/',
  exchange: {
    name: 'logs',
    routingKey: 'json-event.oit-scribe',
    properties: { type: 'topic', durable: true, autoDelete: false }
  },
  application: 'oit-scribe',
  tags: [ 'bunyan', 'node' ]
});
```

With that, I get a successful connection to my RabbitMQ server, my `.on('connect')` callback fires as I would expect, and all log info passes to the proper RabbitMQ exchange.

When I upgrade to v1.0.6, using the exact same code, the connection never succeeds; I just see my `.on('close')` callback fire every second until I kill the app.

If I change the configuration properties on the exchange to include `noDeclare: true`, then the connection to the server itself succeeds, but my connection to the exchange fails; my `.on('connect')` callback never fires, and no log info passes to the exchange.

I tried to figure this one out as best as I could, but can't seem to find _any_ way to understand what the underlying problem is. How can I help you debug this?

**Comment by cp2587** [comment]
I have the exact same problem. Did you manage to find a workaround ?

**Comment by brandonhamilton** [comment]
I have released 1.0.7, which has just updated the dependency to amqp@0.2.6, and this issue no longer appears to occur. Please do verify if this is solved for your use-case.
